src/go/debug-collector/dumper/dumper.go

```diff
@@ -0,0 +1,222 @@
+package dumper
+
+import (
+	"archive/tar"
+	"bytes"
+	"compress/gzip"
+	"encoding/json"
+	"log"
+	"os"
+	"os/exec"
+	"strings"
+
+	"github.com/pkg/errors"
+	corev1 "k8s.io/api/core/v1"
+)
+
+// Dumper struct is for dumping cluster
+type Dumper struct {
+	cmd       string
+	resources []string
+	namespace string
+	location  string
+	Errors    map[string]string
+	mode      int64
+}
+
+// New return new Dumper object
+func New(location, namespace, resource string) Dumper {
+	directory := "cluster-dump"
+	if len(location) > 0 {
+		directory = location + "/cluster-dump"
+	}
+	resources := []string{
+		"pods",
+		"replicasets",
+		"deployments",
+		"statefulsets",
+		"replicationcontrollers",
+		"events",
+		"configmaps",
+		"secrets",
+		"cronjobs",
+		"jobs",
+		"podsecuritypolicies",
+		"poddisruptionbudgets",
+		"perconaxtradbbackups",
+		"perconaxtradbclusterbackups",
+		"perconaxtradbclusterrestores",
+		"perconaxtradbclusters",
+		"clusterrolebindings",
+		"clusterroles",
+		"rolebindings",
+		"roles",
+		"storageclasses",
+		"persistentvolumeclaims",
+		"persistentvolumes",
+	}
+	if len(resource) > 0 {
+		resources = append(resources, resource)
+	}
+	return Dumper{
+		cmd:       "kubectl",
+		resources: resources,
+		location:  directory,
+		Errors:    make(map[string]string),
+		mode:      int64(0777),
+		namespace: namespace,
+	}
+}
+
+type k8sPods struct {
+	Items []corev1.Pod `json:"items"`
+}
+
+type namespaces struct {
+	Items []corev1.Namespace `json:"items"`
+}
+
+// DumpCluster create dump of a cluster in Dumper.location
+func (d *Dumper) DumpCluster() error {
+	tarFile, err := os.Create(d.location + ".tar.gz")
+	if err != nil {
+		return errors.Wrap(err, "create tar file")
+	}
+	defer tarFile.Close()
+	zr := gzip.NewWriter(tarFile)
+	tw := tar.NewWriter(zr)
+	defer zr.Close()
+	defer tw.Close()
```

Comment:
All Close()'s errors should be checked and logged

Reply:
https://github.com/percona/percona-toolkit/pull/452/commits/118110a671ae050d9ba09342b631a0fecf9a964e